tiatoolbox/tools/stainextract.py

```diff
@@ -29,23 +29,78 @@
 from tiatoolbox.utils.misc import get_luminosity_tissue_mask
 
 
+def vectors_in_right_direction(e_vectors):
+    """Points the eigen vectors in the right direction.
+
+    Args:
+        e_vectors (ndarray): eigen vectors
+
+    Returns:
+        ndarray pointing in the correct direction
+
+    """
+    if e_vectors[0, 0] < 0:
+        e_vectors[:, 0] *= -1
+    if e_vectors[0, 1] < 0:
+        e_vectors[:, 1] *= -1
+
+    return e_vectors
+
+
+def h_and_e_in_right_order(v1, v2):
+    """Rearranges input vectors for H&E in correct order with H as first output.
+
+    Args:
+        v1 (ndarray): Input vector for stain extraction.
+        v2 (ndarray): Input vector for stain extraction.
+
+    Returns:
+        input vectors in the correct order.
+
+    """
+    if v1[0] > v2[0]:
+        he = np.array([v1, v2])
+    else:
+        he = np.array([v2, v1])
+
+    return he
+
+
+def dl_output_for_h_and_e(dictionary):
+    """Return correct value for H and E from dictionary learning output.
+
+    Args:
+        dictionary (ndarray): sklearn DictionaryLearning output.
```

Comment:
Would be good to put a link to sklearn DL function:

https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.DictionaryLearning.html

Reply:
Added link in this commit https://github.com/TIA-Lab/tiatoolbox/commit/537e07483777d0a52e0be05dc00c6772277975dd